wsmaster/che-core-api-factory-github-common/src/main/java/org/eclipse/che/api/factory/server/github/AbstractGithubURLParser.java

```diff
@@ -78,6 +79,8 @@ public abstract class AbstractGithubURLParser {
     this.apiClient = githubApiClient;
     this.disableSubdomainIsolation = disableSubdomainIsolation;
     this.providerName = providerName;
+    this.isGitHubServer =
+        !isNullOrEmpty(oauthEndpoint) && !GITHUB_SAAS_ENDPOINT.equals(oauthEndpoint);
```

Comment:
I struggle to understand this line. Could it be extracted to a separate method with Javadoc?

Reply:
Added some javadoc above the variable